image/nitf.ksy

```diff
@@ -0,0 +1,766 @@
+      - id: desid
+        type: str
+        size: 25
+      - id: desver
+        type: str
+        size: 2
+      - id: declasnfo
+        type: clasnfo
+  data_sub_header:
+    instances:
+      tre_ofl:
+        value: des_base.desid == 'TRE_OVERFLOW'
+    params:
+      - id: total_size
+        type: u2
+    seq:
+      - id: des_base
+        type: data_sub_header_base
+      - id: overflowed_header_type
+        -orig-id: desoflw
+        type: str
+        size: 6
+        if: tre_ofl
+      - id: data_item_overflowed
+        -orig-id: desitem
+        type: str
+        size: 3
+        if: tre_ofl
+      - id: des_defined_subheader_fields_len
+        -orig-id: desshl
+        type: str
+        size: 4
+      - id: desshf
+        type: str
+        size: des_defined_subheader_fields_len.to_i
+      - id: des_defined_data_field
+        -orig-id: desdata
+        type: str
+        size: >
+          total_size - (2 + 25 + 2 + des_base.declasnfo._sizeof + 
+            (tre_ofl ? 6 + 3 : 0) + 4 + des_defined_subheader_fields_len.to_i)
+  data_sub_header_tre:
+    seq:
+      - id: des_base
+        type: data_sub_header_base
+      - id: overflowed_header_type
+        -orig-id: desoflw
+        type: str
+        size: 6
+        if: des_base.desid == 'TRE_OVERFLOW'
+      - id: data_item_overflowed
+        -orig-id: desitem
+        type: str
+        size: 3
+        if: des_base.desid == 'TRE_OVERFLOW'
+      - id: des_defined_subheader_fields_len
+        -orig-id: desshl
+        type: str
+        size: 4
+      - id: des_defined_data_field
+        -orig-id: desdata
+        type: str
+        size: des_defined_subheader_fields_len.to_i
+  data_sub_header_streaming:
+    seq:
+      - id: des_base
+        type: data_sub_header_base
+      - id: des_defined_subheader_fields_len
+        -orig-id: desshl
+        type: str
+        size: 4
+      - id: sfh_l1
+        type: str
+        size: 7
+      - id: sfh_delim1
+        type: u4
+      - id: sfh_dr
+        type: u1
+        repeat: expr
+        repeat-expr: sfh_l1.to_i
+      - id: sfh_delim2
+        type: u4
+      - id: sfh_l2
+        type: str
+        size: 7
+  reserved_extension_segment:
+    params:
+      - id: idx
+        type: u2
+    seq:
+      - id: reserved_sub_header
+        type: reserved_sub_header(_parent.header.lrnfo[idx].lrsh.to_i)
+      - id: reserved_data_field
+        size: _parent.header.lrnfo[idx].lr.to_i
+  reserved_sub_header:
+    params:
+      - id: total_size
+        type: u2
+    seq:
+      - id: re
+        contents: 'RE'
+      - id: resid
+        type: str
+        size: 25
+      - id: resver
+        type: str
+        size: 2
+      - id: reclasnfo
+        type: clasnfo
+      - id: resshl
+        type: str
+        size: 4
+      - id: resshf
+        type: str
+        size: resshl.to_i
+      - id: resdata
+        type: str
+        size: total_size - (2 + 25 + 2 + reclasnfo._sizeof + 4 + resshl.to_i)

```

Comment:
@rmspeers Is there any reason not to use [`size-eos: true`](https://doc.kaitai.io/user_guide.html#_variable_length_structures) instead? It is intended specifically for this use case. I can add some quick suggestions for you to see what I mean.